api/v1beta1/glanceapi_types.go

```diff
@@ -107,6 +107,10 @@ type GlanceAPISpec struct {
 	// +kubebuilder:validation:Optional
 	// ExtraMounts containing conf files and credentials
 	ExtraMounts []GlanceExtraVolMounts `json:"extraMounts"`
+
+	// +kubebuilder:validation:Optional
+	// NetworkAttachmentDefinitions list of network attachment definitions the service pod gets attached to
+	NetworkAttachmentDefinitions []string `json:"networkAttachmentDefinitions"`
```

Comment:
So, not in the main `Glance` type, but in the `GlanceAPI` CR, if we add something like:

```
spec:
  ...
  ...
  networkAttachmentDefinitions:
      - internalapi
      - storage
  ...
  ...
```

we should be able to include the pre-existing NADs to the resulting POD.
@stuggi am I understanding it right?

Reply:
@stuggi Maybe we can add a section in the `README` documenting how to set `NADs` and provide an example in the `samples/` directory.

Reply:
right, with the current  version you specify it per api instance, like in your example. works nice for me to attaching glance to a ceph deployed via https://github.com/openstack-k8s-operators/install_yamls/pull/54 . If you want to test it, I have a customized openstack-operator at `OPENSTACK_IMG=quay.io/mschuppe/openstack-operator-index:e2b9953ad06cdc2fc937aa9e5ac57e2bfef4141f` and used to following modifications:

```
     template:
       databaseInstance: openstack
       containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
-      storageClass: ""
-      storageRequest: 10G
+      customServiceConfig: |
+        [DEFAULT]
+        enabled_backends = default_backend:rbd
+        debug = true
+        [glance_store]
+        default_backend = default_backend
+        [default_backend]
+        rbd_store_ceph_conf = /etc/ceph/ceph.conf
+        store_description = "RBD backend"
+        rbd_store_pool = images
+        rbd_store_user = admin
+      extraMounts:
+      - extraVol:
+        - extraVolType: Ceph
+          mounts:
+          - mountPath: /etc/ceph
+            name: ceph
+            readOnly: true
+          propagation:
+          - Glance
+          volumes:
+          - name: ceph
+            projected:
+              sources:
+              - secret:
+                  name: ceph-conf-files
       glanceAPIInternal:
         containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
+        networkAttachmentDefinitions:
+        - storage
       glanceAPIExternal:
         containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
+        networkAttachmentDefinitions:
+        - storage
```

created an image as
```
$ openstack image list
$ curl -L -o /tmp/cirros-0.5.2-x86_64-disk.img http://download.cirros-cloud.net/0.5.2/cirros-0.5.2-x86_64-disk.img
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   273  100   273    0     0    475      0 --:--:-- --:--:-- --:--:--   474
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 15.5M  100 15.5M    0     0  3595k      0  0:00:04  0:00:04 --:--:-- 4928k

$ qemu-img convert -O raw /tmp/cirros-0.5.2-x86_64-disk.img /tmp/cirros-0.5.2-x86_64-disk.img.raw
$ openstack image create --container-format bare --disk-format raw --file /tmp/cirros-0.5.2-x86_64-disk.img.raw cirros

$ openstack image list
+--------------------------------------+--------+--------+
| ID                                   | Name   | Status |
+--------------------------------------+--------+--------+
| 1c876c6a-046e-4f1b-9e66-e8895a58d142 | cirros | active |
+--------------------------------------+--------+--------+

$ oc rsh ceph
sh-4.4# rbd -p images list
1c876c6a-046e-4f1b-9e66-e8895a58d142
```

I plan to update the README of  this PR to show this example.

Reply:
ack and thanks, I guess (but that could be a follow up) that we can introduce the `networkAttachmentDefinition` at the top level and make the split (internal vs external) for the sub CRs.
FWIW I tried your PR and I can reach the `GlanceAPI` pod from the `MetalLB` endpoint.